ext/json/json.c

```diff
@@ -141,24 +142,33 @@ static PHP_MINFO_FUNCTION(json)
 }
 /* }}} */
 
-PHP_JSON_API int php_json_encode_ex(smart_str *buf, zval *val, int options, zend_long depth) /* {{{ */
+PHP_JSON_API int php_json_encode_ex(smart_str *buf, zval *val, int options, zend_long depth, int indent) /* {{{ */
```

Comment:
I'm not so sure where these `PHP_JSON_API` functions are being used and as a result I didn't get to test the changes for functions `php_json_encode_ex` and `php_json_encode`

Reply:
Presumably these are for extension usage, though I'm not sure where they actually get used. I'd suggest accepting `zend_string *` as indent here, as the more general choice than an int.

Reply:
Changed the indent param to `zend_string *`  and allowed it to be empty. If it's empty, the function will generate the default indentation string and free it afterwards.